File: format_queries.js
const fs = require('fs');
const path = 'C:\\Users\\Heota\\.gemini\\antigravity-ide\\brain\\21ca6da9-1afe-4819-b27a-1cb3c4d8338f\\report_uc_data_correction.md';
let content = fs.readFileSync(path, 'utf8');
const regex = /^\d+\.\s+([^:\n]+):\s*\n```sql\n([\s\S]*?)\n```/gm;
const newContent = content.replace(regex, (match, title, query) => {
    const formattedTitle = title.charAt(0).toUpperCase() + title.slice(1);
    return `**${formattedTitle}:**\n${query.trim()}\n`;
});
fs.writeFileSync(path, newContent, 'utf8');
console.log('Done');

File: format_queries.py
import re

file_path = r'C:\Users\Heota\.gemini\antigravity-ide\brain\21ca6da9-1afe-4819-b27a-1cb3c4d8338f\report_uc_data_correction.md'

with open(file_path, 'r', encoding='utf-8') as f:
    content = f.read()

# Regex to match:
# 1. Something:
# `sql
# QUERY
# `
pattern = r'(?m)^\d+\.\s+([^:\n]+):\s*\n`sql\n(.*?)\n`'

def replacer(match):
    title = match.group(1).strip()
    # Capitalize first letter of title if not already
    title = title[0].upper() + title[1:]
    query = match.group(2).strip()
    return f"**{title}:**\n{query}\n"

new_content = re.sub(pattern, replacer, content, flags=re.DOTALL)

with open(file_path, 'w', encoding='utf-8') as f:
    f.write(new_content)

print('Done rewriting queries.')
